commands/inventory/add_service_haproxy.go

```diff
@@ -0,0 +1,95 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package inventory
+
+import (
+	"github.com/percona/pmm/api/inventorypb/json/client"
+	"github.com/percona/pmm/api/inventorypb/json/client/services"
+
+	"github.com/percona/pmm-admin/commands"
+)
+
+var addHAProxyServiceResultT = commands.ParseTemplate(`
+HAProxy Service added.
+Service ID     : {{ .Service.ServiceID }}
+Service name   : {{ .Service.ServiceName }}
+Node ID        : {{ .Service.NodeID }}
+Environment    : {{ .Service.Environment }}
+Cluster name   : {{ .Service.Cluster }}
+Replication set: {{ .Service.ReplicationSet }}
+Custom labels  : {{ .Service.CustomLabels }}
+`)
+
+type addHAProxyServiceResult struct {
+	Service *services.AddHAProxyServiceOKBodyHaproxy `json:"haproxy"`
+}
+
+func (res *addHAProxyServiceResult) Result() {}
+
+func (res *addHAProxyServiceResult) String() string {
+	return commands.RenderTemplate(addHAProxyServiceResultT, res)
+}
+
+type addHAProxyServiceCommand struct {
+	ServiceName    string
+	NodeID         string
+	Environment    string
+	Cluster        string
+	ReplicationSet string
+	CustomLabels   string
+}
+
+func (cmd *addHAProxyServiceCommand) Run() (commands.Result, error) {
+	customLabels, err := commands.ParseCustomLabels(cmd.CustomLabels)
+	if err != nil {
+		return nil, err
+	}
+
+	params := &services.AddHAProxyServiceParams{
+		Body: services.AddHAProxyServiceBody{
+			ServiceName:    cmd.ServiceName,
+			NodeID:         cmd.NodeID,
+			Environment:    cmd.Environment,
+			Cluster:        cmd.Cluster,
+			ReplicationSet: cmd.ReplicationSet,
+			CustomLabels:   customLabels,
+		},
+		Context: commands.Ctx,
+	}
+
+	resp, err := client.Default.Services.AddHAProxyService(params)
+	if err != nil {
+		return nil, err
+	}
+	return &addHAProxyServiceResult{
+		Service: resp.Payload.Haproxy,
+	}, nil
+}
+
+// register command
+var (
+	AddHAProxyService  = new(addHAProxyServiceCommand)
+	AddHAProxyServiceC = addServiceC.Command("haproxy", "Add an haproxy service to inventory").Hide(hide)
+)
+
+func init() {
+	AddHAProxyServiceC.Flag("name", "HAProxy service name. Required").Required().StringVar(&AddHAProxyService.ServiceName)
+	AddHAProxyServiceC.Flag("node-id", "HAProxy service node ID. Required").Required().StringVar(&AddHAProxyService.NodeID)
+	AddHAProxyServiceC.Flag("environment", "Environment name").StringVar(&AddHAProxyService.Environment)
+	AddHAProxyServiceC.Flag("cluster", "Cluster name").StringVar(&AddHAProxyService.Cluster)
+	AddHAProxyServiceC.Flag("replication-set", "Replication set name").StringVar(&AddHAProxyService.ReplicationSet)
+	AddHAProxyServiceC.Flag("custom-labels", "Custom user-assigned labels").StringVar(&AddHAProxyService.CustomLabels)
```

Comment:
I don't want to be picky. it is that I just finished PMM-7473 and in that ticket I was asked to improve the help.
For example, in custom label I added `Example: region=east,app=app1` and for environment we could add an example like 'prod or qa'.
Also, we can use PlaceHolder to show the examples in the help.

Reply:
Yes, I can do same for HAProxy once will be your PR approved. To do that in same way.

Reply:
@percona-csalguero Changed help in style of https://github.com/percona/pmm-admin/pull/136/files